### PR TITLE
General maintenance for frames

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Builds upon the changes in https://github.com/tgstation/tgstation/pull/81477

**1. Qol**
 - Adds detailed examines & screentips for building & deconstructing both machine & computer frames
 - Adding a circuitboard from a rped to a computer frame will automatically screw it in place like before

**2. Code Improvements**
- Merged procs like `update_path_names()` , `get_requested_amt()` etc into their required places to reduce proc calls
- Autodocs procs like `add_cable()`, `add_glass()` etc. makes them private
- Moved code for machine frame into its own file

## Changelog
:cl:
qol: adds examines & screentips for building & deconstructing both machine & computer frames.
qol: Adding a circuitboard from a rped to n computer frame will automatically screw it in place like before.
code: merged procs for computer & machine frames. autodocs them where possible.
code: moved code for machine frame into its own file.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/buildandrepair.dm [hunk 1]
```diff
@@ -2,248 +2,355 @@
...
+		if(state == COMPUTER_FRAME_SCREEN_INSTALLED)
+			new /obj/item/shard(location)
+			new /obj/item/shard(location)
+	..()
+
+/obj/structure/frame/computer/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = NONE
+	if(isnull(held_item))
+		context[SCREENTIP_CONTEXT_ALT_LMB] = "Rotate"
+		return
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:27]: This should be handled by `/datum/component/simple_rotation`
```suggestion
	if(isnull(held_item))
		return
```

File: code/__DEFINES/construction/structures.dm [hunk 2]
```diff
@@ -62,7 +62,14 @@
...
 #define GEAR_LOOSE 2
 
-// Stationary gas tanks
+//Stationary gas tanks
 #define TANK_FRAME 0
 #define TANK_PLATING_UNSECURED 1
 
+//Computer frames construction states
+#define COMPUTER_FRAME_UNANCHORED 0
+#define COMPUTER_FRAME_ANCHORED 1
```
Comment (MrMelbert) [inline code/__DEFINES/construction/structures.dm:71]: You shouldn't need a state for unanchored and anchored, that's already tracked by whether the computer frame is... anchored or unanchored

File: code/game/machinery/computer/buildandrepair.dm [hunk 3]
```diff
@@ -2,248 +2,355 @@
...
-				if(P.use_tool(src, user, 20, volume=50))
-					to_chat(user, span_notice("You wrench the frame into place."))
-					set_anchored(TRUE)
-					state = 1
-				return
-			if(P.tool_behaviour == TOOL_WELDER)
-				if(!P.tool_start_check(user, amount=1))
-					return
+		if(COMPUTER_FRAME_UNANCHORED)
+			if(default_unfasten_wrench(user, tool, 20) == SUCCESSFUL_UNFASTEN)
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:137]:
```suggestion
			if(default_unfasten_wrench(user, tool, 2 SECONDS) == SUCCESSFUL_UNFASTEN)
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 4]
```diff
@@ -2,248 +2,355 @@
...
-				if(istype(P, /obj/item/stack/cable_coil))
-					cable = P
-				else if(istype(P, /obj/item/storage/part_replacer))
-					cable = locate(/obj/item/stack/cable_coil) in P.contents
-				if(!cable)
-					return
+	if(state == COMPUTER_FRAME_UNANCHORED)
+		if(!tool.tool_start_check(user, amount=1))
+			return
+		if(!tool.use_tool(src, user, 20, volume = 50))
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:152]:
```suggestion
		if(!tool.use_tool(src, user, 2 SECONDS, volume = 50))
```

**Review by MrMelbert — COMMENTED** [review]
I also have a branch refactoring computer and machine frames so this may get awkward for me

**Comment by SyncIt21** [comment]
> I also have a branch refactoring computer and machine frames so this may get awkward for me

Any conflicts let me know.  We can work something out 👍

**Comment by MrMelbert** [comment]
I'll put up my PR and we can compare, I know mine doesn't have any examine hint or screentips

**Comment by SyncIt21** [comment]
Drafting till yours get merged so I can build on top of your changes.

Atleast the examines and screentips are mine so I can wait

File: code/game/machinery/computer/buildandrepair.dm [hunk 5]
```diff
@@ -83,6 +205,8 @@
 			return ITEM_INTERACT_BLOCKING
 
 /obj/structure/frame/computer/crowbar_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:208]: I explicitly avoided doing this, to let people bash frames with crowbars while in combat mode (like jaws of life or something) 

...But I realize I only did it for two states. (`FRAME_COMPUTER_STATE_BOARD_SECURED` and `FRAME_COMPUTER_STATE_WIRED`)

File: code/game/machinery/computer/buildandrepair.dm [hunk 6]
```diff
@@ -22,12 +109,49 @@
...
+						user.Beam(src, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+				return TRUE
+
+			return FALSE
 
 /obj/structure/frame/computer/item_interaction(mob/living/user, obj/item/tool, list/modifiers, is_right_clicking)
 	. = ..()
 	if(. & ITEM_INTERACT_ANY_BLOCKER)
-		return .
+		return
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:154]: We keep this as per style guide https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-
```suggestion
		return .
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 7]
```diff
@@ -9,6 +9,93 @@
...
+
+	switch(state)
+		if(FRAME_COMPUTER_STATE_EMPTY)
+			if(held_item.tool_behaviour == TOOL_WRENCH)
+				context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Un" : ""]anchor"
+				return CONTEXTUAL_SCREENTIP_SET
+			if(anchored)
+				if(istype(held_item, /obj/item/circuitboard/computer))
+					context[SCREENTIP_CONTEXT_LMB] = "Install board"
+					return CONTEXTUAL_SCREENTIP_SET
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:42]:
```suggestion
			if(anchored && istype(held_item, /obj/item/circuitboard/computer))
				context[SCREENTIP_CONTEXT_LMB] = "Install board"
				return CONTEXTUAL_SCREENTIP_SET
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 8]
```diff
@@ -48,12 +172,10 @@
 			if(istype(tool, /obj/item/storage/part_replacer))
 				return install_parts_from_part_replacer(user, tool) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
 
-	return .
-
 /obj/structure/frame/computer/screwdriver_act(mob/living/user, obj/item/tool)
 	. = ..()
-	if(. != NONE)
-		return .
+	if(. & ITEM_INTERACT_SUCCESS)
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:177]: If you are changing this I would use
```suggestion
	if(. & ITEM_INTERACT_ANY_BLOCKER)
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 9]
```diff
@@ -48,12 +172,10 @@
...
+		return
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:178]: Same as the other one
```suggestion
		return .
```

File: code/game/machinery/constructable_frame.dm [hunk 10]
```diff
@@ -97,14 +97,13 @@
...
 
 /obj/structure/frame/wrench_act(mob/living/user, obj/item/tool)
+	. = NONE
 	switch(default_unfasten_wrench(user, tool, 4 SECONDS))
 		if(SUCCESSFUL_UNFASTEN)
 			return ITEM_INTERACT_SUCCESS
 		if(FAILED_UNFASTEN)
 			return ITEM_INTERACT_BLOCKING
 
-	return NONE
```
Comment (MrMelbert) [inline code/game/machinery/constructable_frame.dm:106]: This seems at tad un-necessary and as per the style guide prefer return over dot usage https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-

File: code/game/machinery/machine_frame.dm [hunk 11]
```diff
@@ -0,0 +1,472 @@
...
+		if(FRAME_STATE_BOARD_INSTALLED)
+			if(held_item.tool_behaviour == TOOL_CROWBAR)
+				context[SCREENTIP_CONTEXT_LMB] = "Pry out components"
+				return CONTEXTUAL_SCREENTIP_SET
+			else if(held_item.tool_behaviour == TOOL_WRENCH)
+				. = NONE
+				if(!circuit.needs_anchored)
+					context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Un" : ""]anchor"
+					. = CONTEXTUAL_SCREENTIP_SET
+				return
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:69]:
```suggestion
				if(!circuit.needs_anchored)
					context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Un" : ""]anchor"
					return CONTEXTUAL_SCREENTIP_SET
				return NONE
```

File: code/game/machinery/machine_frame.dm [hunk 12]
```diff
@@ -0,0 +1,472 @@
...
+/obj/structure/frame/machine/can_be_unfasten_wrench(mob/user, silent)
+	. = ..()
+	if(. != SUCCESSFUL_UNFASTEN)
+		return .
+
+	if(circuit?.needs_anchored)
+		balloon_alert(user, "frame must be anchored!")
+		return FAILED_UNFASTEN
+
+	return
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:286]:
```suggestion
	return .
```

File: code/game/machinery/machine_frame.dm [hunk 13]
```diff
@@ -0,0 +1,472 @@
...
+	return ITEM_INTERACT_BLOCKING
+
+/obj/structure/frame/machine/wirecutter_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(state != FRAME_STATE_WIRED)
+		return NONE
+
+	balloon_alert(user, "removing cables...")
+	if(!tool.use_tool(src, user, 2 SECONDS, volume = 50) || state != FRAME_STATE_WIRED)
+		return
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:307]:
```suggestion
	if(state != FRAME_STATE_WIRED)
		return NONE

	balloon_alert(user, "removing cables...")
	if(!tool.use_tool(src, user, 2 SECONDS, volume = 50) || state != FRAME_STATE_WIRED)
		return ITEM_INTERACT_BLOCKING
```
Same as before with dot memes

File: code/game/machinery/machine_frame.dm [hunk 14]
```diff
@@ -0,0 +1,472 @@
...
+
+	state = FRAME_STATE_EMPTY
+	update_appearance(UPDATE_ICON_STATE)
+	new /obj/item/stack/cable_coil(drop_location(), 5)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/structure/frame/machine/crowbar_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:315]:
```suggestion
```
And same here (though this is redundant anyways? Unless it's for runtime protection? If so, should be commented)

File: code/game/machinery/machine_frame.dm [hunk 15]
```diff
@@ -0,0 +1,472 @@
...
+	dump_contents()
+	balloon_alert(user, "circuit board[length(leftover_components) ? " and components" : ""] removed")
+	// Circuit exited handles updating state
+	return ITEM_INTERACT_SUCCESS
+
+/**
+ * Attempts to add the passed part to the frame
+ * Requires no sanity check that the passed part is a stock part
+ * Arguments
+ *
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:330]: Very minor nitpick but this messes up the spacing when it gets autodoc'd
```suggestion
 * Attempts to add the passed part to the frame
 *
 * Requires no sanity check that the passed part is a stock part
 *
 * Arguments
```

File: code/game/machinery/machine_frame.dm [hunk 16]
```diff
@@ -0,0 +1,472 @@
...
+				return install_circuit_from_part_replacer(user, tool) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
+
+		if(FRAME_STATE_BOARD_INSTALLED)
+			if(istype(tool, /obj/item/storage/part_replacer))
+				return install_parts_from_part_replacer(user, tool) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
+
+			if(!user.combat_mode)
+				return add_part(user, tool) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
+
+	return
```
Comment (MrMelbert) [inline code/game/machinery/machine_frame.dm:430]:
```suggestion
	return  .
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 17]
```diff
@@ -111,38 +235,32 @@
...
-			return FALSE
-
-		if(FRAME_COMPUTER_STATE_WIRED)
-			var/obj/item/stack/sheet/glass/glass_sheets = locate() in replacer
-			if(isnull(glass_sheets))
-				return FALSE
+/obj/structure/frame/computer/wirecutter_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(state != FRAME_COMPUTER_STATE_WIRED)
+		return
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:241]:
```suggestion
	if(state != FRAME_COMPUTER_STATE_WIRED)
		return ITEM_INTERACT_BLOCKING
```

File: code/game/machinery/computer/buildandrepair.dm [hunk 18]
```diff
@@ -22,12 +109,49 @@
...
 		balloon_alert(user, "circuit already installed!")
 		return FALSE
-	return ..()
+	if(!anchored && istype(board))
+		balloon_alert(user, "frame must be anchored!")
+		return FALSE
+	. = ..()
+	if(. && !by_hand)
+		state = FRAME_COMPUTER_STATE_BOARD_SECURED
+		update_appearance(UPDATE_ICON_STATE)
```
Comment (MrMelbert) [inline code/game/machinery/computer/buildandrepair.dm:118]:
```suggestion
	if(. && !by_hand) // Installing via RPED auto-secures it
		state = FRAME_COMPUTER_STATE_BOARD_SECURED
		update_appearance(UPDATE_ICON_STATE)
	return . 
```
- more dot memes
- Just a comment to explain what's going on, not immediately obvious

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**